lib/hanami/middleware/body_parser/form_parser.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require_relative "parser"
+require "rack/multipart"
+
+module Hanami
+  module Middleware
+    class BodyParser
+      # @since 2.0.0
+      # @api private
+      class FormParser < Parser
+        # @since 2.0.0
+        # @api private
+        MIME_TYPES = [
+          "application/x-www-form-urlencoded",
```

Comment:
@jodosha I think we don't need this parser for the mimetype `application/x-www-form-urlencoded`. `multipart/form-data` should be enough in my opinion. 

Otherwise we may get the following `params`-results, when `:form`-parser is used, because  `Rack::Multipart.parse_multipart(env)` returns `nil` for a ``application/x-www-form-urlencoded` payload.

```console
{
  :framework => "hanami",
  :is        => "great",
  :_         => nil
}
```
 Apart from that the branch works like a charm. Now files are uploaded correctly. Why the test still fails is another story that we will clear up.

Reply:
@wuarmin This is taken from Rack codebase.